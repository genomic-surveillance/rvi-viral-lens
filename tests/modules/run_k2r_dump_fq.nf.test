nextflow_process {

    name "Test Process run_k2r_dump_fq process"
    script "modules/run_kraken2ref_and_pre_report.nf"
    process "run_k2r_dump_fastqs_and_pre_report"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                meta = [
                    id:"test.9000000",
                    sample_id:"test",
                    taxid:9000000
                ]
                test_loc="${projectDir}/tests/test_data/"
            }

            process {

            """
            // define inputs of the process here. Example:
            input[0] = tuple(
                params.meta,
                [
                    file("${params.test_loc}/test_kraken_outputs/test.class_seqs_1.fq"),
                    file("${params.test_loc}/test_kraken_outputs/test.class_seqs_2.fq")
                ],
                file("${params.test_loc}/test_kraken_outputs/test.9000000_tax_to_reads.json"),
                file("${params.test_loc}/test_kraken_outputs/test.9000000_decomposed.json"),
                file("${params.test_loc}/test_kraken_outputs/test.report.txt")
                )
            """

            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }
}
