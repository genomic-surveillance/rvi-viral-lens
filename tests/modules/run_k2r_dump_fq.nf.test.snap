{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test.9000000",
                            "taxid": 9000000,
                            "sample_id": "test"
                        },
                        [
                            "test.9000000_2697049_R1.fq:md5,b0eaac69db411379704837a2b2a9dbd8",
                            "test.9000000_2697049_R2.fq:md5,705ce6f753733d8ba3ccdffad193ef52",
                            "test.9000000_9000019_R1.fq:md5,5618fdbb4f73478ee97133395db9ec44",
                            "test.9000000_9000019_R2.fq:md5,e7cd0de58458d603335bbc239e5cfafd"
                        ]
                    ]
                ],
                "1": [
                    "test.9000000_pre_report.tsv:md5,b4650118fe4fccc3b649658761c454f2"
                ],
                "fq_files": [
                    [
                        {
                            "id": "test.9000000",
                            "taxid": 9000000,
                            "sample_id": "test"
                        },
                        [
                            "test.9000000_2697049_R1.fq:md5,b0eaac69db411379704837a2b2a9dbd8",
                            "test.9000000_2697049_R2.fq:md5,705ce6f753733d8ba3ccdffad193ef52",
                            "test.9000000_9000019_R1.fq:md5,5618fdbb4f73478ee97133395db9ec44",
                            "test.9000000_9000019_R2.fq:md5,e7cd0de58458d603335bbc239e5cfafd"
                        ]
                    ]
                ],
                "report_file": [
                    "test.9000000_pre_report.tsv:md5,b4650118fe4fccc3b649658761c454f2"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-07-24T18:02:50.990401"
    }
}